mlpipeline/src/mlpipeline/etl/pipeline.py

```
@@ -16,47 +17,29 @@ def __init__(self, conn: psycopg2.extensions.connection, parser: IngredientParse
-    def process_recipe(self, line: str) -> tuple[int, Optional[Exception]]:
-        try:
-            recipe_data = Recipe.model_validate_json(line)
+            recipe_json = scrape_recipe(recipe_url)
+            recipe_data = RawRecipe.model_validate_json(recipe_json)
+            recipe_id, err = self.process_recipe(recipe_data)
+            self.create_recipe_embeddings_batch([{
+                'id': recipe_id,
+                'text': recipe_json
+            }])
+            if err is not None:
+                raise err
```

Ruff (0.14.13)

Abstract `raise` to an inner function

(TRY301)